### ISSUE TITLE
`prefer-destructuring` flags a property assignment

### ISSUE THREAD

**Issue by bfred-it** [eslint/eslint#11584]
_Edit: read this comment instead https://github.com/eslint/eslint/issues/11584#issuecomment-479892994_

**Tell us about your environment**

* **ESLint Version:** 5.16
* **Node Version:** any, happens on repl
* **npm Version:** any, happens on repl

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 9,
        "sourceType": "script",
        "ecmaFeatures": {}
    },
    "rules": {
        "prefer-destructuring": 2
    },
    "env": {}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const arr = ['value'];

exports.value = arr[0];
```

https://eslint.org/demo/#eyJ0ZXh0IjoiY29uc3QgYXJyID0gWyd2YWx1ZSddO1xuXG5leHBvcnRzLnZhbHVlID0gYXJyWzBdOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6OSwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJwcmVmZXItZGVzdHJ1Y3R1cmluZyI6Mn0sImVudiI6e319fQ==

**What did you expect to happen?**

No `prefer-destructuring` error should appear

**What actually happened? Please include the actual, raw output from ESLint.**

```
Use array destructuring
```

**Comment by g-plane** [comment]
Duplicate of #11581

**Comment by bfred-it** [comment]
Uhm, it's not. The issue isn't the number, but that the destructuring suggested **is not possible at all**

**Comment by platinumazure** [comment]
Reopening as this is, at very least, not a duplicate.

I did look on MDN and see nothing about destructured assignment into an object property. I don't know enough about ES2015 to say if this is definitely a bug, but I can't rule it out at this point.

**Comment by bfred-it** [comment]
My bad, that's actually possible: 🤭

```js
// Error: Use array destructuring
exports.value = arr[0];

// Valid, but... I don't know how understandable this is
[exports.value] = arr;
```

So... technically this isn't a bug. Sorry about that.

However perhaps this should be its own option. This is the closest that I can get to disabling the behavior, but I think it also disables regular declarations like `const [x] = arr`

```
'prefer-destructuring': [
	'error',
	{
		VariableDeclarator: {
			array: false,
			object: true
		}
	}
}
```

**Comment by platinumazure** [comment]
@bfred-it Thanks for your response. I would say in this case, there is no bug in the rule.

Please feel free to open an enhancement request if you think the rule could be improved.

Closing as ESLint is working as intended here.
